naptime-graphql/src/main/scala/org/coursera/naptime/ari/graphql/schema/NaptimeTopLevelResourceField.scala

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright 2016 Coursera Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.coursera.naptime.ari.graphql.schema
+
+import com.linkedin.data.DataMap
+import com.typesafe.scalalogging.StrictLogging
+import org.coursera.naptime.ResourceName
+import org.coursera.naptime.ari.graphql.SangriaGraphQlContext
+import org.coursera.naptime.schema.Handler
+import org.coursera.naptime.schema.HandlerKind
+import org.coursera.naptime.schema.Resource
+import sangria.schema.Argument
+import sangria.schema.Context
+import sangria.schema.Field
+import sangria.schema.ObjectType
+
+object NaptimeTopLevelResourceField extends StrictLogging {
+
+  val MUTATION_HANDLERS: Set[HandlerKind] = Set(
+    HandlerKind.ACTION,
+    HandlerKind.CREATE,
+    HandlerKind.DELETE,
+    HandlerKind.PATCH,
+    HandlerKind.UPSERT)
+
+  /**
+    * Generates an object-type for a given resource name, with each field on the merged output
+    * schema available on this object-type.
+    *
+    * @param resource Resource to generate lookup type for
+    * @return WithSchemaErrors[ObjectType] including the ObjectType for the resource,
+    *         if we were able to generate it, and any errors generated while creating the type.
+    */
+  def generateLookupTypeForResource(
+      resource: Resource,
+      schemaMetadata: SchemaMetadata):
+    WithSchemaErrors[Option[ObjectType[SangriaGraphQlContext, DataMap]]] = {
+
+    val resourceName = ResourceName.fromResource(resource)
+
+    val fieldsAndErrors = resource.handlers
+      .filterNot(handler => MUTATION_HANDLERS.contains(handler.kind))
+      .map { handler =>
+        handler.kind match {
+          // We want to make sure that if a resource has a GET handler, it also has a MULTI_GET
```

Comment:
Is placing this comment here still relevant? it seems like we've moved this logic down to `generateGetHandler`